### ISSUE TITLE
seem an extra highlight space in the end of each overlength line

### ISSUE THREAD

**Issue by roachsinai** [tjdevries/overlength.vim#6]
![image](https://user-images.githubusercontent.com/9500049/85952873-cfc53280-b99e-11ea-82e5-23f310866c18.png)

**Comment by tjdevries** [comment]
Do you happen to have any mutli-byte characters on those lines? I wasn't able to reproduce on my machine.

Can you give me version of vim/neovim you're using and link to config files? I'll see if I can figure it out.

**Comment by roachsinai** [comment]
No, when I open help page, comes too.

![image](https://user-images.githubusercontent.com/9500049/85973399-29f8de80-ba05-11ea-85b4-2ac47a37c161.png)

I'm using gVim but always use in terminal.

```
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Jun 06 2020 14:24:18)
包含补丁: 1-914
编译者 Arch Linux
巨型版本 with GTK3 GUI.
```

I'll check does it caused by conflict across plugins first. Will feed back later.

**Comment by roachsinai** [comment]
```
call plug#begin('~/.vim/plugged')
Plug 'tjdevries/overlength.vim'
call plug#end()


let overlength#default_overlength = 79

call overlength#disable_filetypes(['markdown', 'text', 'git', 'floggraph', ''])

call overlength#set_overlength('vim', 120)
```

Hi, I just use setting above and get pic below:

![image](https://user-images.githubusercontent.com/9500049/85974344-d76cf180-ba07-11ea-983c-f8660d3f9e6c.png)

Check the help page, and the `print` line in below buffer.

**Comment by roachsinai** [comment]
Hi @tjdevries I found for me overlength.vim matched newline.

I'm on Manjaro using vim.

**Comment by tjdevries** [comment]
Oh, very interesting! Can you try this match in your setup:

```
/\%>100v[^\n]*
```

(Replace 100 with whatever you want your textwidth to be)

**Comment by tjdevries** [comment]
If that fixes it, I can make the change required and push it.

**Comment by roachsinai** [comment]
Yeah, it works! As I very new to highlight. Just wonder `.*` should not match `\n` as I know.

**Comment by tjdevries** [comment]
Yeah, I didn't expect them to match. Does the latest commit work for you?

**Comment by roachsinai** [comment]
> Oh, very interesting! Can you try this match in your setup:
> 
> ```
> /\%>100v[^\n]*
> ```
> 
> (Replace 100 with whatever you want your textwidth to be)

No, maybe need one more `/` in the beginning of match pattern compare you posted yesterday.

**Comment by tjdevries** [comment]
Try one more time? Just pushed a new change.

**Comment by roachsinai** [comment]
No, maybe my response confused you, 😃.

`/\%>100v[^\n]*` this did work, which you told me yesterday!
Today's first commit got `\%>100v[^\n]*` which no leading `/` compare to yesterday.
And Today's second commit got
```
\%>100v[^
]*
```
still not work.

I think maybe just need a leading `/`.

And if you want you could rebase this commit to one, I'll reset and pull it.

As the commit is too new to effect another people using this plugin.

**Comment by roachsinai** [comment]
```
function! s:get_repeat_char() abort
  return get(g:, 'overlength#highlight_to_end_of_line', v:true) ? '[^\n]*' : ''
endfunction
```
with

```
      let w:overlength_pattern = '/\%' .
```

works for me now.

**Comment by tjdevries** [comment]
well you can set `overlength#highlight_to_end_of_line` to `v:false` and then it will be that value I think.

**Comment by tjdevries** [comment]
Hmmm, I'm confused i think. I can look tomorrow, getting to sleepy 😆

**Comment by roachsinai** [comment]
Yeah, good night!

**Comment by tjdevries** [comment]
Closed by #7
